### PR TITLE
feat: Add SESSION_DOMAIN env var

### PR DESCRIPTION
**Pull request by asetty**
Allow the user to customize the Domain attribute for user
session / cookie.

By default the value is "" and will default to the URL used by the
oidc provider when making the request to the callback.

Signed-off-by: Adam Setters <asetty@arista.com>

### CODE REVIEW

**Comment by asetty** [comment]
@yanniszark could you take a look at this? 

Also heads up, I'll be sending a PR to add functionality to include the scheme and host for the  origURL state sometime this week.

**Comment by yanniszark** [comment]
@asetty thanks for the PR and sorry for the wait.
I'm rather busy with Kubecon these days.
I would feel much more comfortable approving changes if we had an issue outlining what the general plan is.
Can you summarize the current state, the problem and the proposed solution in https://github.com/arrikto/oidc-authservice/issues/43?
Please help me see the bigger picture :)
